openstack/sharedfilesystems/v2/schedulerstats/results.go

```diff
@@ -0,0 +1,120 @@
+package schedulerstats
+
+import (
+	"encoding/json"
+	"math"
+
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+// Capabilities represents the information of an individual Pool.
+type Capabilities struct {
+	// The following fields should be present in all storage drivers.
+
+	// The quality of service (QoS) support.
+	Qos string `json:"qos"`
+	// The date and time stamp when the API request was issued.
+	Timestamp string `json:"timestamp"`
+	// The name of the share back end.
+	ShareBackendName string `json:"share_backend_name"`
+	// The mapping between servers and pools.
+	ServerPoolsMapping string `json:"server_pools_mapping"`
+	// Share server is usually a storage virtual machine or a lightweight container that is used to export shared file systems.
+	DriverHandlesShareServers string `json:"driver_handles_share_servers"`
+	// The driver version of the back end.
+	DriverVersion string `json:"driver_version"`
+	// The amount of free capacity for the back end, in GiBs. A valid value is a string, such as unknown, or an integer.
+	FreeCapacityGB float64 `json:"-"`
+	// The storage protocol for the back end. For example, NFS_CIFS, glusterfs, HDFS, etc.
+	StorageProtocol string `json:"storage_protocol"`
+	// The total capacity for the back end, in GiBs. A valid value is a string, such as unknown, or an integer.
+	TotalCapacityGB float64 `json:"-"`
+	// The specification that filters back ends by whether they do or do not support share snapshots.
+	SnapshotSupport string `json:"snapshot_support"`
+	// The back end replication domain.
+	ReplicationDomain string `json:"replication_domain"`
+	// The name of the vendor for the back end.
+	VendorName string `json:"vendor_name"`
+
+	// The following fields are optional and may have empty values depending
+
+	// on the storage driver in use.
+	ReservedPercentage  int64   `json:"reserved_percentage"`
+	AllocatedCapacityGB float64 `json:"-"`
+}
+
+// Pool represents an individual Pool retrieved from the
+// schedulerstats API.
+type Pool struct {
+	// The pools for the back end.
+	Pools string `json:"pools"`
+	// The name of the back end.
+	Name string `json:"name"`
+	// The name of the back end.
+	Backend string `json:"backend"`
+	// The pool name for the back end.
+	Pool string `json:"pool"`
+	// The host name for the back end.
+	Host string `json:"host"`
+	// The back end capabilities which include qos, total_capacity_gb, etc.
+	Capabilities Capabilities `json:"capabilities"`
```

Comment:
The capabilities are only returned when querying for the pool details. I believe we should have a separate PoolDetails struct.

Reply:
Other details endpoint don't do it like this and only have one result. I am also not sure how that would even be done because it is always the generic paginator object returned.